pkg/drivers/generic/generic.go

```diff
@@ -25,6 +26,40 @@ const (
 	defaultMaxIdleConns = 2 // copied from database/sql
 )
 
+type Config struct {
+	MetricsRegisterer    prometheus.Registerer
+	Endpoint             string
+	Scheme               string
+	DataSourceName       string
+	ConnectionPoolConfig ConnectionPoolConfig
+	BackendTLSConfig     tls.Config
+}
+
+type Constructor func(ctx context.Context, cfg *Config) (leaderElect bool, backend server.Backend, err error)
```

Comment:
Might the registry better fit in `pkg/drivers`? This would make the calls in other packages a little more idiomatic - `drivers.Register()`, `drivers.SetDefault()/GetDefault()`, `drivers.Resolve()` or perhaps `drivers.ForScheme()` etc.